solr/core/src/java/org/apache/solr/update/DocumentBuilder.java

```diff
@@ -161,14 +162,21 @@ public static Document toDocument(
           && !sfield.multiValued()
           && field.getValueCount() > 1
           && !(sfield.getType() instanceof DenseVectorField)) {
+
+        // Ensure we do not flood the logs with extremely long values
+        String fieldValue = field.getValue().toString();
+        if (fieldValue.length() > MAX_VALUES_AS_STRING_LENGTH) {
+          fieldValue = fieldValue.substring(0, MAX_VALUES_AS_STRING_LENGTH - 4) + "...]";
```

Comment:
Not sure about that trailing bracket

Reply:
Okay; that fieldValue is guaranteed to already end in that bracket is not obvious. Adding a simple `assert fieldValue.endsWith(']');` before truncating would be simple enough to clarify.